crates/starknet-devnet-core/src/predeployed_accounts.rs

```diff
@@ -85,138 +81,3 @@ impl AccountGenerator for PredeployedAccounts {
         Ok(&self.accounts)
     }
 }
-
-#[cfg(test)]
-mod tests {
-    use jsonschema::JSONSchema;
-    use serde_json::json;
-    use starknet_types::contract_class::Cairo0Json;
-    use starknet_types::felt::{Felt, Key};
-    use starknet_types::traits::{ToDecimalString, ToHexString};
-
-    use crate::constants::{CAIRO_0_ACCOUNT_CONTRACT_HASH, CAIRO_0_ACCOUNT_CONTRACT_PATH};
-    use crate::predeployed_accounts::PredeployedAccounts;
-    use crate::traits::AccountGenerator;
-    use crate::utils::test_utils::dummy_contract_address;
-
-    const SEED: u32 = 123;
-
-    const PRIVATE_KEYS_IN_HEX: [&str; 3] = [
-        "0xc4da537c1651ddae44867db30d67b366",
-        "0xd6a82a951b923e0a443cdef36840ff07",
-        "0x610e4ad509c47055dff4948fe6b4f832",
-    ];
-
-    const PUBLIC_KEYS_IN_HEX: [&str; 3] = [
-        "0x60dea6c1228f1db4ca1f9db11c01b6e9cce5e627f7181dcaa27d69cbdbe57b5",
-        "0x5a91f0ea25312accb20d8041b12260bff31a3490e5730a690b0ec8fe10ffbb",
-        "0x2b45af00df833ea1a4895c49a18ebd84309b79d658cae05e274a7b1cac47016",
-    ];
-
-    const PREDEPLOYED_ACCOUNTS_JSON_SCHEMA_WITH_DATA_FILE_PATH: &str =
-        concat!(env!("CARGO_MANIFEST_DIR"), "/schemas/predeployed_accounts_fixed_seed.json");
-
-    /// Test correct generation of private keys
-    /// Test expected results are taken from https://github.com/0xSpaceShard/starknet-devnet/blob/master/test/support/schemas/predeployed_accounts_fixed_seed.json
-    #[test]
-    fn correct_private_key_from_fixed_seed() {
-        let predeployed_acc = predeployed_account_instance();
-        let generated_private_keys = predeployed_acc.generate_private_keys(3).unwrap();
-
-        let expected_result = PRIVATE_KEYS_IN_HEX
-            .into_iter()
-            .map(|hex_str| Felt::from_prefixed_hex_str(hex_str).unwrap())
-            .collect::<Vec<Felt>>();
-
-        assert_eq!(expected_result, generated_private_keys);
-
-        let string_result = generated_private_keys
-            .into_iter()
-            .map(|pk| pk.to_prefixed_hex_str())
-            .collect::<Vec<String>>();
-
-        assert_eq!(string_result, PRIVATE_KEYS_IN_HEX.to_vec());
-    }
-
-    #[test]
-    fn private_key_from_different_seeds_should_be_different() {
```

Comment:
Couldn't this test be preserved if two different seeds were used? In fact couldn't we keep all tests except `correct_private_key_from_fixed_seed`?

Reply:
the test can be preserved. Others rely on the above mentioned seed.

Reply:
That's what I meant. That is the question: do we want to modify the old tests so that they are still applicable. But isn't e.g. `check_generated_predeployed_accounts_against_json_schema` still usable? Doesn't it only check against the schema?

Reply:
It checks the values inside.

Reply:
Ok, I was confused by `assert!(schema.is_valid, ...)`, but now I see what it does.

Reply:
the mentioned test is recovered in [dbef561](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/324/commits/dbef561b2df9905c7a50ac9d213afd7ffd6513d1)

Reply:
I'm suspicious about these non-deterministic tests that don't loop for a large number of times. But if you don't want to waste your time on that anymore, feel free to resolve this conversation and merge (you already have one approval).

Reply:
[e669df5](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/324/commits/e669df5416390307364cc2cd7d2c3038a9a11ed3)